src/mixing.html

```diff
@@ -15,53 +15,78 @@ <h2 id="mixing">Annotating MathML</h2>
+  reference such as <code>$name</code> refereces a descendent element
+  that has an attriute <code>arg="name"</code>. unlike <code class="attribute">id</code>
+  attributes, <code class="attribute">arg</code> do not have to be
+  unique within a document, when searching for a matching element the
+  search should only consider descendns and not consider descendents
+  of any element that has an <code class="attribute">intent</code> or
+  <code class="attribute">arg</code> attribute.</li>
+  <li><em>literal numbers</em> such as <code>2.5</code> denote
+  themselves. <span class="ednote">0.25e1?</span></li>
+  <li><em>literal identifiers</em> Non numeric literals reference
+  terms in the intent dictionary, currently located at <a
+  href="https://docs.google.com/spreadsheets/d/1EsWou1K5nxBdLPvQapdoA9h-s8lg_qjn8fJH64g9izQ/edit#gid=1358098730">Google
+  Sheets</a>. Note the list of [=intent=] identifiers is
+  <em>Open</em>: level 1 is a curated list maintained by the Math Working
+  Group and should be implemented by any implemenation of MathML
+  intent, so that suitable audio renderings of the level 1 terms will
+  be generated. Terms in other levels may be contributed; the
+  dictionary may give speech hints, but a system may also fall back on
+  reading the identifier name as given.</li>
+  <li><em>the wildcard</em>, <code>*</code> denotes all the children
+  <span class="ednote">just mi and mn children?</span> of the element so
+  that <code>intent="somefunction(*)"</code> is equivalent to
+  <code>intent="somefunction($1,$2, ...)"</code></li>
+  </ul>
+
+<p>The key component here is the literal; they are intended to correspond to some mathematical concept or operator, or some application specific quantity or operation; that is it represents some “meaning”. Ultimately, the goal is to translate this virtual content tree into text or speech (in different languages), braille or some other form. The preferred translation may also depend on context and user profile. On the one hand, the most natural translation of a given expression can depend on the operator. It is thus desirable to have a set of common known meaning with translation rules. Thus power(x,2) might be rendered as “the 2nd power of x” or “x squared”, depending on the user and agent.</p>
+
   <section>
    <h4 id="mixing_intent_grammar">The Grammar for <code class="attribue">intent</code></h4>
 
    <pre>
-intent := literal | selector | intent '(' wildcard ')' | intent '(' intent [ ',' intent ]* ')'
-literal      := [letters|digits|_|-]*
-selector     := ( argref | numref ) [ '/' selector ]?
-argref       := '$' NCName
-numref       := '$' [digit]+
-wildcard     := '@'
+intent   := literal | selector | intent '(' wildcard ')' | intent '(' intent [ ',' intent ]* ')'
+literal  := ( letters | digits | '.' | '_' | '-' )*
+selector := argref | numref
+argref   := '$' NCName
+numref   := '$' [digit]+ ( '/' numref )?
+wildcard := '*'
```

Comment:
I understand this is meant to help with the n-ary infix cases, but *does it* help with the n-ary infix cases? I'm not so sure, given that it doesn't mean "all children", but "all children that the author meant were arguments".

Does it matter if one does:
```xml
<mrow intent="cross-product(*)">
  <mo>(</mo>
  <mi>A</mi>
  <mo>×</mi>
  <mi>B</mi>
  <mo>×</mi>
  <mi>C</mi>
  </mo>)</mo>
</mrow>
```

vs
```xml
<mo>(</mo>
<mrow intent="cross-product(*)">
  <mi>A</mi>
  <mo>×</mi>
  <mi>B</mi>
  <mo>×</mi>
  <mi>C</mi>
</mrow>
<mo>)</mo>
```

or is it *only* going to work on arg-annotated trees?
```xml
<mrow intent="cross-product(*)">
  <mo>(</mo>
  <mi arg="A">A</mi>
  <mo>×</mi>
  <mi arg="B">B</mi>
  <mo>×</mi>
  <mi arg="C">C</mi>
  </mo>)</mo>
</mrow>
```

I am unconvinced it is really needed at this point in time.

Reply:
Me too:-) I think that's the most "at risk" part of the grammar, it is temping but specifying what exactly it picks up is problematical. That's really why I am pushing to get a grammar in the spec, at least then we all look at exactly the same version and if it's wrong we can change it.

I think that it probably has to mean "any mi or mn child" so all three of  the cases you show work, but if you add any redundant mrows it doesn't match. But maybe that doesn't work well enough but if we end up with some complicated algorithm to say what it should select then it should go.  Again I'd like to see some specifications of default intent with and without this.